lib/jsonapi/include_directive.rb

```diff
@@ -40,6 +42,29 @@ def [](key)
       end
     end
 
+    # @param another_directive [IncludeDirective]
+    # @return [IncludeDirective]
+    def merge(other)
+      fail ArgumentError,
+           "parameter MUST be an IncludeDirective" unless
+          other.is_a?(IncludeDirective)
+
+      hash = to_hash.dup
+      Parser.deep_merge!(hash, other.to_hash)
```

Comment:
I'm actually storing the result after duplicating it with `to_hash.dup` to specifically avoid this issue 😃
However, after a careful read I noticed that `#to_hash` should already return a copy, not the original reference. So probably calling `#dup` on it is not really needed